File: Python/Strings/1262 - Leitura Múltipla.py
"""
Autor: Pierre Vieira
Data da submissão: 08/02/2020 21:05:44
"""


def solve_this_problem():
    string = ''
    cont = 0
    for s in rastro:
        if s == 'W':
            if len(string) > 0:
                cont += 1
            cont += 1
            string = ''
        else:
            string += s
            if p == len(string):
                cont += 1
                string = ''
    if len(string) > 0:
        cont += 1
    return cont


while True:
    try:
        rastro = input()
    except EOFError:
        break
    else:
        p = int(input())
        print(solve_this_problem())

File: Kotlin/Strings/ 1262 - Leitura Múltipla.kt
/*
* Autor: Pierre Vieira
* Data de submissão: 12/03/2023 00:11:36
*/

fun main(args: Array<String>) {
    while (true) {
        val rastro = readLine() ?: break
        val p = readLine()?.toInt() ?: break
        println(solve(rastro, p))
    }
}

fun solve(rastro: String, p: Int): Int {
    var string = ""
    var cont = 0
    rastro.forEach {
        if (it == 'W') {
            if (string.isNotEmpty()) {
                cont++
            }
            cont++
            string = ""
        } else {
            string += it
            if (string.length == p) {
                cont++
                string = ""
            }
        }
    }
    if (string.isNotEmpty()) {
        cont++
    }
    return cont
}